mda_openbabel_converter/OpenBabelParser.py

```diff
@@ -3,27 +3,286 @@
+        MDAnalysis Topology?)
         """
-        pass
+        if HAS_OBABEL == False:
+            return False
+        else:
+            return isinstance(thing, ob.OBMol)
 
     def parse(self, **kwargs):
         """
         Accepts an OpenBabel OBMol and returns a MDAnalysis Topology. Will need
         to extract the number of atoms, number of residues, number of segments,
-        atom_residue index, residue_segment index and other attributes from the
-        OBMol to initialise a new Topology. 
+        atom_residue index, residue_segment index and all of the atom's
+        relevant attributes from the OBMol to initialise a new Topology.
         """
-        pass
\ No newline at end of file
+        mol = self.filename
+
+        self.atoms = []
+        self.n_atoms = 0
+        self.residues = []
+        self.n_residues = 0
+        self.segments = []
+        self.n_segments = 0
+
+        # Atoms
+        names = []
+        chiralities = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        aromatics = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        icodes = []
+        occupancies = []
+        tempfactors = []  # B factor; not supported by OB
+
+        if mol.Empty():
+            return Topology(n_atoms=0,
+                            n_res=0,
+                            n_seg=0,
+                            attrs=None,
+                            atom_resindex=None,
+                            residue_segindex=None)
+
+        for atom in ob.OBMolAtomIter(mol):
+            # need to add handling incase attributes are invalid or null in OBMol
+            # names.append(atom.GetType()) #char -> nothing for name in OBMol? Is name required to make MDA Atom?
+            atomtypes.append(atom.GetType())  # char
+            ids.append(atom.GetIdx()) #int
+            masses.append(atom.GetExactMass())  # double -> what about atom.GetAtomicMass()??; which is better?
+            if abs(atom.GetExactMass()-atom.GetAtomicMass()) >= NEUTRON_MASS:
+                warnings.warn(
+                    f"Exact mass and atomic mass of atom ID: {atom.GetIdx()}"
+                    " are more than 1.008 AMU different. Be aware of isotopes," 
+                    " which are NOT supported by MDAnalysis.")
+            charges.append(atom.GetPartialCharge()) #int (or use atom.GetFormalCharge()?)
+
+            # convert atomic number to element
+            elements.append(OBElementTable().GetSymbol(atom.GetAtomicNum())) #char
+
+            if atom.HasResidue():
+                resid = atom.GetResidue() # null if no residue
+                resnums.append(resid.GetNum()) # TO DO: check if start at 0 or 1
+                resnames.append(resid.GetName())
+                chainids.append(resid.GetChainNum()) # is this correct???
```

Comment:
Looking at the existing parsers in MDAnalysis and the test code, chainID should be a string/char (e.g. 'A' for chain A). The `GetChain()` method in Open Babel seems like the correct method for this.
![image](https://github.com/user-attachments/assets/5fd5f6ea-8160-4462-81e9-77f64751784a)
See: https://userguide.mdanalysis.org/stable/topology_system.html
and https://openbabel.org/api/3.0/classOpenBabel_1_1OBResidue.shtml#acb948f24a90d8d093a8b3149cd07cdb4